app/src/utils/spatial-utils.tsx

```diff
@@ -178,23 +84,78 @@ export const parseSpatialDataByType = (
   return { markerLayers: [occurrencesMarkerLayer], staticLayers: [boundaryStaticLayer] };
 };
 
-// checks which key should be used to identify the layer
-export const getFeatureLayerKey = (spatialRecord: ISpatialData, feature: Feature): string => {
-  let key = '';
-
-  if (isOccurrenceFeature(feature)) {
-    key = `${spatialRecord.associated_taxa}`;
+const occurrenceMarkerSetup = (
+  latLng: LatLngTuple,
+  taxaData: ITaxaData[],
+  datasetVisibility: IDatasetVisibility
+): IMarker | null => {
+  const submission_ids: number[] = taxaData
+    .filter((item: ITaxaData) => {
+      if (item.associated_taxa) {
+        return datasetVisibility[item.associated_taxa] === undefined ? true : datasetVisibility[item.associated_taxa];
+      }
+      return false;
+    })
+    .map((item: ITaxaData) => item.submission_spatial_component_id);
+
+  if (submission_ids.length > 0) {
+    return {
+      position: latLng,
+      key: submission_ids.join('-'),
+      popup: <FeaturePopup submissionSpatialComponentIds={submission_ids} />,
+      count: submission_ids.length
+    };
   }
 
-  if (isBoundaryFeature(feature)) {
-    key = `${spatialRecord.submission_spatial_component_id}`;
-  }
+  return null;
+};
 
-  if (isBoundaryCentroidFeature(feature)) {
-    key = `${spatialRecord.submission_spatial_component_id}`;
-  }
+export const parseProjectResults = (data: ISpatialData[], datasetVisibility: IDatasetVisibility): ISearchResult[] => {
+  const results: ISearchResult[] = [];
+  data.forEach((item) => {
+    if (item.spatial_data.features[0]) {
+      if (item.spatial_data.features[0].properties) {
+        if (item.spatial_data.features[0].properties.type === SPATIAL_COMPONENT_TYPE.BOUNDARY_CENTROID) {
```

Comment:
Consider using the existing `isBoundaryCentroidFeature` function at the bottom of this file, as it already does a similar check.